recipes-core/images/core-image-minimal.bbappend

```diff
@@ -3,3 +3,15 @@ tune_ext4fs () {
 }
 
 IMAGE_CMD:ext4 = "oe_mkext234fs ext4 ${EXTRA_IMAGECMD}; tune_ext4fs"
+
+clean_etc_systemd_stuff() {
+    # we want /etc/systemd/system clean, so move stuff to /lib/systemd/system
+    cp -a ${IMAGE_ROOTFS}${sysconfdir}/systemd/system/* ${IMAGE_ROOTFS}${systemd_unitdir}/system
```

Comment:
Maybe you should move only service files which are *not* symlinks, if I remember correctly enabling a service just creates a symlink from `/lib/system/system/` --> `/etc/systemd/system`. 

This could lead to disabled services, which might not be desired.

Reply:
According to my understanding of the systemd directory handling, files that ship (in packages) from the distribution should go into /usr/lib/systemd/. Modifications done by system administrator (user) go into /etc/systemd/system/.
Yocto runs some magic during the rootfs generation which (accidentially) creates symlinks/install files below /etc. This is why I moved all out of this directory without deeper inspection. Symlinks from /lib to /etc are by definition not allowed according to my understanding, so in case this happens we have a bigger issue to solve :-)